### PR TITLE
[PM-10365] Drop OrganizationUser AccessAll

### PR DESCRIPTION
**Pull request by eliykat**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-10365
## 📔 Objective

Final step in removing OrganizationUser.AccessAll column:

* MSSQL: (1) add a default column value, (2) update sprocs to remove the column, (3) drop the column
  * also drop some old sprocs no longer in use and the TVP they used
* EF: remove shadow property configuration and run migrations

**WARNING:** this includes an index rebuild on the `OrganizationUsers` table, because the index previously included the `AccessAll` column. I have dropped & recreated it without this column on the assumption it's still useful for the remaining columns in the index. However we may need to coordinate this index rebuild with BRE.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: util/Migrator/DbScripts/2024-09-03_02_DropOrganizationUserAccessAll_DropColumn.sql [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        [dbo].[OrganizationUser]
+    DROP COLUMN
+        [AccessAll]
+END
+GO
+
+-- Recreate the index without the column
+PRINT N'Recreating index IX_OrganizationUser_UserIdOrganizationIdStatus...';
+CREATE NONCLUSTERED INDEX [IX_OrganizationUser_UserIdOrganizationIdStatus]
+    ON [dbo].[OrganizationUser]([UserId] ASC, [OrganizationId] ASC, [Status] ASC);
```
Comment (eliykat) [inline util/Migrator/DbScripts/2024-09-03_02_DropOrganizationUserAccessAll_DropColumn.sql:38]: Note the index rebuild

File: util/Migrator/DbScripts/2024-09-03_00_DropOrganizationUserAccessAll_DefaultColumnValue.sql [hunk 2]
```diff
@@ -0,0 +1,11 @@
+-- Finalise removal of OrganizationUser.AccessAll column
+-- Add default column value
+-- Sprocs already have default value for rollback purposes, this just supports dropping the column itself
+
+IF OBJECT_ID('[dbo].[DF_OrganizationUser_AccessAll]', 'D') IS NULL
+  AND COL_LENGTH('[dbo].[OrganizationUser]', 'AccessAll') IS NOT NULL
```
Comment (eliykat) [inline util/Migrator/DbScripts/2024-09-03_00_DropOrganizationUserAccessAll_DefaultColumnValue.sql:6]: @rkac-bw the equivalent migraiton for `Groups` didn't check for the existence of the column (only for the non-existence of the constraint), so it throws an error if run multiple times. Theoretically dbup shouldn't run a migration multiple times, but it is something we try to support. Do you want me to go back and fix that?

Compare: https://github.com/bitwarden/server/blob/main/util/Migrator/DbScripts/2024-09-02_00_DropGroupAccessAll_DefaultColumnValue.sql#L5

**Review by rkac-bw — DISMISSED** [review]
lgtm

**Review by rkac-bw — APPROVED** [review]
LGTM

**Pull request merged**